src/components/ArticlePage.jsx

```diff
@@ -1,64 +1,49 @@
-import React from 'react';
-import { useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
-import { fetchSingleArticle,fetchArticleComments } from '../ApiRequests';
-import './ArticlePage.css';
-
-// complete is error state 
+import { fetchSingleArticle, updateArticleVote } from "../ApiRequests";
+import "./ArticlePage.css";
+import { Comments, Votes } from "./index";
+// todo - complete is error state
 
 function ArticlePage() {
-  const { article_id } = useParams(); 
- const [isCommentsVisible, setCommentsVisible] = useState(false)
-  const [article, setArticle] = useState({})
-  const [comments, setComments] = useState([])
-  const [isLoading, setIsLoading] = useState(true)
- 
-  
-    useEffect(() => {
-      fetchSingleArticle(article_id)
-        .then((articleFromApi) => {
-        setArticle(articleFromApi)
-        setIsLoading(false)
-      });
-    }, []);
-  
+  const { article_id } = useParams();
+  const [article, setArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+
   useEffect(() => {
-    fetchArticleComments(article_id)
-      .then((commentsFromApi) => {
-     setComments(commentsFromApi);
-   
-   });
+    fetchSingleArticle(article_id).then((articleFromApi) => {
+      setArticle(articleFromApi);
+      setIsLoading(false);
+    });
   }, []);
-  
+
+  const handleVote = (articleId, vote) => {
+    updateArticleVote(articleId, vote).then((updatedArticleFromApi) => {
+      setArticle(updatedArticleFromApi);
```

Comment:
Here you are setting the votes to be the number of votes the articles has after patching. This will include all of the votes made since you first fetched it and isn't always a change of 1.

Instead of using the actual number of votes updated the state to show the change the user would expect, i.e. +1 or -1